drivers/at24cxxx/at24cxxx.c

```diff
@@ -0,0 +1,248 @@
+/**
+ * @ingroup     drivers_at24cxxx
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the AT24C128 / AT24C256 EEPROM units.
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+
+#define ENABLE_DEBUG  (0)
+#include "debug.h"
+#include "assert.h"
+#include "xtimer.h"
+#include "byteorder.h"
+
+#include "include/at24cxxx_params.h"
+#include "include/at24cxxx_defines.h"
+#include "at24cxxx.h"
+
+static int _read(const at24cxxx_t *dev, uint32_t pos, uint8_t *data, size_t len)
+{
+    int check;
+    uint8_t polls = AT24CXXX_MAX_POLLS;
+
+    while ((check =
+                i2c_read_regs(dev->params.i2c, dev->params.dev_addr, pos, data,
+                              len,
+                              I2C_REG16)) == -ENXIO && --polls) {
+        xtimer_usleep(AT24CXXX_POLL_DELAY_MS * US_PER_MS);
+    }
```

Comment:
No need to squeeze everything into the while condition, I think it's much more readable this way:

```suggestion
    while ((check = i2c_read_regs(dev->params.i2c,
                                  dev->params.dev_addr,
                                  pos, data, len, I2C_REG16)) == -ENXIO) {
            if (--polls == 0) {
                break;
            }
            xtimer_usleep(AT24CXXX_POLL_DELAY_MS * US_PER_MS);
    }
```

Reply:
Ok